src/devices/src/legacy/serial.rs

```diff
@@ -346,52 +372,100 @@ mod tests {
 
         let mut serial = Serial::new_out(intr_evt, Box::new(serial_out.clone()));
 
+        // Invalid write of multiple chars at once.
         serial.write(u64::from(DATA), &[b'x', b'y']);
-        serial.write(u64::from(DATA), &[b'a']);
-        serial.write(u64::from(DATA), &[b'b']);
-        serial.write(u64::from(DATA), &[b'c']);
+        // Valid one char at a time writes.
+        RAW_INPUT_BUF
+            .iter()
+            .for_each(|&c| serial.write(u64::from(DATA), &[c]));
         assert_eq!(
-            serial_out.buf.lock().unwrap().as_slice(),
-            &[b'a', b'b', b'c']
+            serial_out.internal.lock().unwrap().write_buf.as_slice(),
+            &RAW_INPUT_BUF
         );
     }
 
     #[test]
-    fn serial_input() {
+    fn serial_raw_input() {
```

Comment:
nit: tests name typically start with `test_` so that you can easily differentiate them from helper functions.

Reply:
Fixed in #1494